stripe/src/main/java/com/stripe/android/checkout/CheckoutViewModel.kt

```diff
@@ -36,13 +40,35 @@ internal class CheckoutViewModel internal constructor(
         mutableTransition.postValue(target)
     }
 
-    fun getPaymentMethods(
-        customerId: String,
+    /**
+     * Fetch the customer's saved payment methods.
+     * The activity args are assumed to be static across the lifetime of the view model
+     * so calling this method multiple times will not result in re-fetching the payment methods.
+     *
+     * If you wish to force an update, call [updatePaymentMethods] directly.
+     */
+    fun getPaymentMethods(activity: Activity): LiveData<List<PaymentMethod>> {
```

Comment:
Wonder if we should refactor this to `updatePaymentMethods()` with no return type and a separate `LiveData` property that can be observed. That better fits my mental model for caching.

Reply:
I guess my thinking here was that you'd only call `updatePaymentMethods` when you know that you want to force an update, and I was trying to keep that logic out of the view. I guess just doing a simple `savedInstanceState == null` check in the view is fine though.

Reply:
hmm `savedInstanceState` appears to always be null. I suppose that makes sense since we're not saving any state (for a second I was thinking there would be some default state).

In any case, I realized this doesn't cover transitioning between fragments, at least given the current setup. If the user taps that "add card" button, and then backs out (returning to the payment methods list), we don't want to re-fetch the list as we know nothing has changed.